src/pcapi/core/offers/api.py

```diff
@@ -183,43 +183,70 @@ def update_offers_active_status(query, is_active):
             redis.add_offer_id(client=app.redis_client, offer_id=offer_id)
 
 
-def create_stock(
+def get_valid_create_stock_model(
     offer: models.Offer,
     price: float,
     quantity: int = None,
     beginning: datetime.datetime = None,
     booking_limit_datetime: datetime.datetime = None,
-) -> Stock:
+):
     """Return the new stock or raise an exception if it's not possible."""
     validation.check_required_dates_for_stock(offer, beginning, booking_limit_datetime)
     validation.check_offer_is_editable(offer)
     validation.check_stocks_are_editable_for_offer(offer)
+    validation.check_stock_price(price)
+    validation.check_stock_quantity(quantity if quantity else 0)
```

Comment:
Est ce que ça serait pas plus lisible d'instancier le stock d'abord et de valider ses attributs ensuite ?
Comme ça tes deux nouvelles validations sont bien des validations sur le modèles (qui peuvent être vérifiées à d'autres moment) contrairement aux trois précédentes qui sont plus des validations sur l'action même de création / modification de stock

```suggestion
    stock = models.Stock(...)
    
    validation.check_stock_price(stock)
    validation.check_stock_quantity(stock)
    
    return stock
```

Reply:
Au contraire, je pense que ça n'a pas de sens d'instancier un modèle de données avec des données incorrectes. Si on peut valider les données avant instanciation (comme Romain l'a fait ici), je pense qu'on devrait le faire.